docs-website/topics/serialization-customization-options.md

```diff
@@ -0,0 +1,519 @@
+    // language is always serialized
+    println(Json.encodeToString(adminUser))
+    // {"name":"Alice","projects":[{"name":"kotlinx.serialization","language":"Kotlin"}]}
+
+    // Excludes projects because it's an empty list
+    // and EncodeDefault.Mode is set to NEVER, so it's not serialized
+    println(Json.encodeToString(guestUser))
+    // {"name":"Bob"}
+}
+//sampleEnd
+```
+{kotlin-runnable="true" id="serialize-encode-default"}
+
+### Make properties required with the `@Required` annotation
+
+Annotate a property with [`@Required`](https://kotlinlang.org/api/kotlinx.serialization/kotlinx-serialization-core/kotlinx.serialization/-required/) to make it required in the input.
+This enforces that the input contains the property, even if it has a [default value](#set-default-values-for-optional-properties):
+
+```kotlin
+// Imports declarations from the serialization library
+import kotlinx.serialization.*
+import kotlinx.serialization.json.*
+import kotlinx.serialization.Required
+
+//sampleStart
+@Serializable
+// Marks the language property as required
+data class Project(val name: String, @Required val language: String = "Kotlin")
+
+fun main() {
+    val data = Json.decodeFromString<Project>("""
+        {"name":"kotlinx.serialization"}
+    """)
+    println(data)
+    // MissingFieldException
+}
+//sampleEnd
+```
+{kotlin-runnable="true" id="required-annotation-example"}
+
+## Customize class serialization
+
+Kotlin serialization provides several ways to modify how classes are serialized.
+This section covers techniques for customizing property names, controlling property inclusion, and more.
+
+### Customize serial names
+
+By default, property names in the serialization output, such as JSON, match their names in the source code.
+
+You can customize these names, called _serial names_,
+with the [`@SerialName`](https://kotlinlang.org/api/kotlinx.serialization/kotlinx-serialization-core/kotlinx.serialization/-serial-name/) annotation.
+Use it to make a property name more descriptive in the serialized output:
+
+```kotlin
+// Imports declarations from the serialization library
+import kotlinx.serialization.*
+import kotlinx.serialization.json.*
+
+//sampleStart
+@Serializable
+// Changes the lang property to language using @SerialName
+class Project(val name: String, @SerialName("language") val lang: String)
+
+fun main() {
+    val data = Project("kotlinx.serialization", "Kotlin")
+    // Prints the more descriptive property name in the JSON output
+    println(Json.encodeToString(data))
+    // {"name":"kotlinx.serialization","language":"Kotlin"}
+}
+//sampleEnd
+```
+{kotlin-runnable="true" id="serial-names"}
+
+### Define constructor properties for serialization
+
+A class annotated with `@Serializable` must declare all parameters in its primary constructor as properties.
+
+If you need to perform additional initialization logic before assigning values to properties, use a secondary constructor.
```

Comment:
Let's link to our content on secondary constructors:
https://kotlinlang.org/docs/classes.html#secondary-constructors